### PR TITLE
patch: Provision flintlock from latest release

### PR DESCRIPTION
**Pull request by Callisto13**
Previously flintlock only had pre-releases, which meant a different
install method in `provision.sh`. Now it can use the standard func.

Also update the compatibility table and add a note about supported
Firecracker versions.

### CODE REVIEW

File: hack/scripts/provision.sh [hunk 1]
```diff
@@ -294,7 +294,7 @@ install_flintlockd() {
 	say "Installing flintlockd version $tag to $INSTALL_PATH"
 
 	if [[ "$tag" == "$DEFAULT_VERSION" ]]; then
-		tag=$(latest_pre_release_tag "$FLINTLOCK_REPO")
+		tag=$(latest_release_tag "$FLINTLOCK_REPO")
```
Comment (yitsushi) [inline hack/scripts/provision.sh:297]: Do we need the `latest_pre_release_tag` function after this patch?

**Review by yitsushi — DISMISSED** [review]

File: hack/scripts/provision.sh (hunk 1 above)
Comment (Callisto13) [reply hack/scripts/provision.sh:297]: technically no. i know i am usually ruthless when it comes to deleting dead code, but for some reason I had great attachment to this one 🤣

Comment (Callisto13) [reply hack/scripts/provision.sh:288]: it's dead i killed it hope you are happy 😭 😭

**Review by richardcase — APPROVED** [review]

File: hack/scripts/provision.sh (hunk 1 above)
Comment (yitsushi) [reply hack/scripts/provision.sh:289]: :scream:

**Pull request merged**